pkg/engine/context/context.go

```diff
@@ -190,7 +190,11 @@ func (ctx *context) AddOldResource(data map[string]interface{}) error {
 }
 
 // AddTargetResource adds data at path: target
-func (ctx *context) AddTargetResource(data map[string]interface{}) error {
+func (ctx *context) SetTargetResource(data map[string]interface{}) error {
+	if err := addToContext(ctx, nil, "target"); err != nil {
+		logger.Error(err, "unable to replace target resource")
+		return err
```

Comment:
In what scenario will the error occur?

Reply:
no idea but `addToContext` returns an error, we have to check it anyway

Reply:
Isn't `addToContext` only merges the new value when the error is nil?

https://github.com/kyverno/kyverno/blob/b3a56176e5c0bc6504a67adb00ed783642598dcc/pkg/engine/context/context.go#L142-L146

Reply:
>That's why we set it to nil before setting the new value (to prevent the merge).

Ok read it again, I think I misunderstood it. Thought we are setting nil to check the error. But it actually resets the target context.

Reply:
yes, it resets the target context, then we set the target to the new values, effectively bypassing the merge.